File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_77.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function prototype
char* floatToString(float num);

int main() {
    srand(time(NULL)); // Seed the random number generator

    float a = ((float)rand() / RAND_MAX);
    float b = ((float)rand() / RAND_MAX);
    float c = ((float)rand() / RAND_MAX);

    char* str_a = floatToString(a);
    char* str_b = floatToString(b);
    char* str_c = floatToString(c);

    printf("str_a: %s\n", str_a);
    printf("str_b: %s\n", str_b);
    printf("str_c: %s\n", str_c);

    // Free allocated memory
    free(str_a);
    free(str_b);
    free(str_c);

    return 0;
}

char* floatToString(float num) {
    char* str = malloc(32 * sizeof(char)); // Allocate memory for the string
    if (str == NULL) {
        fprintf(stderr, "Memory allocation failed\n");
        exit(EXIT_FAILURE);
    }
    snprintf(str, 32, "%f", num); // Convert float to string with a precision of 6 decimal places
    return str;
}

File: samples_generated_code/sample_2/phi4/baseline_code_java/code_row_77.java
import java.util.Random;

public class RandomFloatToString {
    public static void main(String[] args) {
        Random random = new Random();
        
        float a = random.nextFloat();
        float b = random.nextFloat();
        float c = random.nextFloat();

        String str_a = convertToDecimalString(a);
        String str_b = convertToDecimalString(b);
        String str_c = convertToDecimalString(c);

        System.out.println("str_a: " + str_a);
        System.out.println("str_b: " + str_b);
        System.out.println("str_c: " + str_c);
    }

    public static String convertToDecimalString(float value) {
        return String.format("%.6f", value); // Formats to 6 decimal places
    }
}